doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,177 @@
+function. This ``subplot`` function is rather unintuitive and makes it
+difficult to specify moderately complex subplots. There are alternatives, some
+of them very powerful, but they generally require a lot of code to specifiy
+precisely the layout. There might be room for an in-between method that rely on
+the graphical description of the layout. It can cover all ``subplot`` cases and
+should allow to specify moderately complex subplots.
+
+Examples
+--------
+
+::
+
+  "AB"
+  ┌────────┐┌────────┐
+  │ A      ││ B      │
+  │        ││        │
+  │        ││        │
+  └────────┘└────────┘
+
+  "ABB"
+  ┌──────┐┌──────────┐
+  │ A    ││ B        │
+  │      ││          │
+  │      ││          │
+  └──────┘└──────────┘
+
+  "ABD"
+  "CCD"
+  ┌───────┐┌───────┐┌───────┐
+  │ A     ││ B     ││ D     │
+  │       ││       ││       │
+  │       ││       ││       │
+  └───────┘└───────┘│       │
+  ┌────────────────┐│       │
+  │ C              ││       │
+  │                ││       │
+  │                ││       │
+  └────────────────┘└───────┘
+
+  "AaBb"
+  ┌───────┐┌─┐┌───────┐┌─┐
+  │ A     ││ ││ B     ││ │
+  │       ││ ││       ││ │
+  │       ││ ││       ││ │
+  └───────┘└─┘└───────┘└─┘
+
+  "  b  "
+  "aABCc"
+              ┌───────┐
+              └───────┘
+  ┌─┐┌───────┐┌───────┐┌───────┐┌─┐
+  │ ││ A     ││ B     ││ C     ││ │
+  │ ││       ││       ││       ││ │
+  │ ││       ││       ││       ││ │
+  └─┘└───────┘└───────┘└───────┘└─┘
+
+
+Syntax
+------
+
+A subplot is described using a capital letter, from ``A`` to ``Z`` such that
+only 26 subplots can be specified at once. Colorbars are specified using
+non-capital letters, from ``a`` to ``z``. The main command is the ``layout``
+command that returns a list of axes (subplots and colorbars) following the
+lexicographic order, mixing capital and small letters. For example::
+
+  A,B,C = layout("ABC")
+  A,B,b = layout("ABb")
+
+A layout can span several lines. In such a case, it is necessary the specify the
+layout using an array of strings::
+
+  A,B = layout(["A",
+                "B"])
+
+The size of a subplot is relative to the size of its string representation, a
+single letter representing the logical unit. For example::
+
+
+  A,B = layout("AAB")
+
+means ``A`` is two logical block wide while ``B`` is one logical block
+wide. The final result should be ``A`` width to be 2/3 while ``B`` width should
+be 1/3.
+
+Colorbars can be specified using a small letter and has either a fixed width
+(vertical colorbar) or a fixed height (horizontal colorbar). For example::
+
+  A,a = layout("Aa")
+
+The exact width or height of colorbars must be specified throught matplotlib
+properties. The difficulty being the attachment of the colorbar to the right
+figure. In the example above, the size of ``A`` should be logical block and the
+colorbar ``a`` should be attached to the right side of ``A``.
+
+  A,a,B = layout("AaB")
```

Comment:
Just so I'm clear, does this mean that the lower case letter is a colorbar that is associated with the corresponding upper case letter?  What happens with "AcB", for example?  The `c` colorbar is orphaned?

Also, I suppose there are some alignment issues with this approach, e.g.:

```
AaB
CDE
```

How would that be laid out?

I see where you are going with using the upper/lower case relationship to associated axes with colorbars.  That's the one part of this spec that I find a bit strange.  It does limit things to ASCII -- but there are so many other things in programming that are limited, at least practically speaking, to ASCII, that this is just "one more thing" -- we probably shouldn't let that stop us.

The only alternative I can think of (and I don't know if it's any better) is to use `v^<>` to associate a colorbar with an axes.  In that case, I'd say we don't need to return the colorbars as axes, since they are accessible through `axes.colorbar` anyway.  Examples:

```
A<B

  v
>ABC<

              ┌───────┐
              └───────┘
  ┌─┐┌───────┐┌───────┐┌───────┐┌─┐
  │ ││ A     ││ B     ││ C     ││ │
  │ ││       ││       ││       ││ │
  │ ││       ││       ││       ││ │
  └─┘└───────┘└───────┘└───────┘└─┘
```